### PR TITLE
syslinux: 2015-11-09 -> 2019-02-07

### PR DESCRIPTION
**Pull request by samueldr**
###### Motivation for this change

Updates to a newer version.

Main important feature:

 * Reproducibility patches

This removes the now unneeded patches. This also switches to the upstream repository instead of a GitHub mirror. Finally, adds myself as maintainer.

###### Things done

- ✔️ Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - ✔️ NixOS
- ✔️ Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- ✔️ Assured whether relevant documentation is up to date
- ✔️ Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @grahamc (reproducibility)

### CODE REVIEW

**Review by grahamc — APPROVED** [review]
Nicely done!

**Comment by vcunat** [comment]
@samueldr: I'm sorry, I just hack-fixed the build now: 5559120.

**Comment by samueldr** [comment]
Don't be sorry. But uh, what issue? I verified the x86_64 linux iso to work as it did with the PR and it worked.

What test did I miss? What issue did I cause?

**Comment by vcunat** [comment]
I think it did work, until another binutils update.

**Comment by samueldr** [comment]
Right, this explains the debian patches which I initially included, until I tested with them all disabled and saw that none were necessary :). Whew, good to know I didn't accidentally the whole build.

File: pkgs/os-specific/linux/syslinux/default.nix [hunk 1]
```diff
@@ -1,31 +1,19 @@
-{ stdenv, fetchFromGitHub, fetchurl, nasm, perl, python, libuuid, mtools, makeWrapper }:
+{ stdenv, fetchFromRepoOrCz, fetchpatch, nasm, perl, python, libuuid, mtools, makeWrapper }:
 
 stdenv.mkDerivation rec {
-  name = "syslinux-2015-11-09";
+  # This is syslinux-6.04-pre3^1; syslinux-6.04-pre3 fails to run.
+  # Same issue here https://www.syslinux.org/archives/2019-February/026330.html
+  name = "syslinux-2019-02-07";
```
Comment (samueldr) [inline pkgs/os-specific/linux/syslinux/default.nix]: `05ac953c23f90b2328d393f7eecde96e41aed067` is `2019-02-20`

**Pull request merged**